### PR TITLE
KFSPTS-14927 only validate date ranges if billing frequency requires it

### PR DESCRIPTION
**Pull request by jhulslander**

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/module/ar/document/web/struts/CuContractsGrantsInvoiceDocumentAction.java [hunk 1]
```diff
@@ -146,6 +147,23 @@ protected void validateBillingPeriodDateRange(ContractsGrantsInvoiceDocument con
             warningMessages.add(configurationService.getPropertyValueAsString(CUKFSKeyConstants.WARNING_CINV_BILLING_PERIOD_START_DATE_BEFORE_AWARD_START_DATE));
         }
     }
+    
+    private boolean shouldValidateBillingPeriodDateRange(ContractsGrantsInvoiceDocument contractsGrantsInvoiceDocument) {
+        String frequencyCode = contractsGrantsInvoiceDocument.getInvoiceGeneralDetail().getAward().getBillingFrequency().getFrequency();
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/module/ar/document/web/struts/CuContractsGrantsInvoiceDocumentAction.java]: Are any null checks or refreshReferenceObject() calls needed on any of the objects in this method chain?

Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/module/ar/document/web/struts/CuContractsGrantsInvoiceDocumentAction.java]: good point, I will investigate

Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/module/ar/document/web/struts/CuContractsGrantsInvoiceDocumentAction.java]: above in the function call, the invoice general detail is referenced from the document.  The getAward function makes a call to Spring to get the award.  So  I think a null check on the award is appropriate.  I also found the frequency code as a member variable on the award, so I don't need to get the frequency object first.

**Comment by jhulslander** [comment]
I've made some changes based on the code review, this is ready for review again

**Comment by jhulslander** [comment]
ah good catch about the invoice general detail, I didn't find that before, I shall switch

**Comment by jhulslander** [comment]
I've made some further changes, this is ready for review again

**Comment by jhulslander** [comment]
this failed due the B2B connection iissue:  Unable to connect to remote site for punchout.

**Comment by jhulslander** [comment]
retest this please

**Comment by jhulslander** [comment]
retest this please

**Pull request merged**